eogrow/pipelines/export_maps.py

```diff
@@ -235,51 +212,142 @@ def _prepare_files(self, geotiff_paths: List[str], output_file: str) -> Tuple[FS
             temp_geotiff_paths,
             workers=None,
             multiprocess=False,
-            desc="Making local copies",
         )
-        temp_map_path = fs.path.basename(output_file)
-        return temp_fs, temp_geotiff_paths, temp_map_path
+        return temp_fs, temp_geotiff_paths
+
+    def _split_patches_temporally(
+        self, filesystem: FS, output_folder: str, geotiff_paths: List[str], some_eopatch: str, crs: CRS
+    ) -> Dict[Optional[dt.datetime], List[str]]:
+        """Splits input patches into multiple tiffs according to the temporal dimension.
+
+        Here the jobs are grouped together per-inputfile, in later parts jobs are grouped per-time.
+        Due to this jobs are created per-input, but we also take note of per-time grouping, which the function returns.
+        """
+        LOGGER.info("Splitting TIFF files temporally.")
+        num_bands, timestamp = self._load_bands_and_timestamp(some_eopatch)  # assume eopatches share these
+
+        def get_extraction_path(input_path: str, time: dt.datetime) -> str:
+            """Ensures that after extraction the files have a time-suffix."""
+            tiff_name = _strip_tiff_extension(fs.path.basename(input_path))
+            return fs.path.join(output_folder, get_tiff_name(self.map_name, tiff_name, crs, time))
+
+        geotiffs_per_time: Dict[Optional[dt.datetime], List[str]] = defaultdict(list)
+        grouped_split_jobs = []  # these are grouped per input file to avoid IO blocking of processes
+        for tiff_path in geotiff_paths:
+            tiff_sys_path = filesystem.getsyspath(tiff_path)
 
-    def _load_timestamp(self, eopatch_name: str) -> List[dt.datetime]:
+            jobs = []
+            for i, time in enumerate(timestamp):
+                extraction_path = get_extraction_path(tiff_path, time)
+                bands = range(i * num_bands, (i + 1) * num_bands)
+                jobs.append(SplitTiffsJob(tiff_sys_path, bands, filesystem.getsyspath(extraction_path)))
+
+                geotiffs_per_time[time].append(extraction_path)
+
+            grouped_split_jobs.append(jobs)
+
+        parallelize(self._execute_split_jobs, grouped_split_jobs, workers=None)
```

Comment:
why is the extra layer in the `grouped_split_jobs` necessary? Why not just run the `get_extraction_path` over _all_ `SplitTiffJobs`?

Reply:
Actually, this extra layer allows you to add the cleanup of the multi-temporal tiff already in the `_execute_split_jobs` in L263, does it not?

Reply:
the extra layer prevents multiple processes to try and extract from the same TIFF. I'm not sure if gdal has parallel TIFF readers, and I don't want to test that shit out.

I could also do the clean-up, yes, but that would require me to do the pickle/unpickle of the filesystem. I try to avoid that since it's an in-house magic spell that I don't fully trust

Reply:
fine as it is, the layering makes sense now.

There are usually also more tiffs than there are timestamps, and more tiffs than there are generally CPU cores, would it make sense to paralellize over tiffs, but to do a sequential run over timestamp extraction? would make the code simpler, not sure about time of processing, but if all CPU cores are used, it shouldn't make a difference, right?

Reply:
isn't that what is currently done? CPUs are given batches of jobs, where each batch pertains to the same tiff, so that only one CPU is processing that batch :thinking: 
You're probably suggesting to skip the extraction to a batch of jobs, and just leave it as a single job (that contains all the extractions), so reshuffling of abstractions. It's not a bad idea, i'll see if the code simplifies a bit with that. I think the main reason that multiple jobs are created (atm) is because we need to produce the output names on the fly. But maybe I can do it in a better way.

Reply:
> isn't that what is currently done?

I checked again and you're right, sorry bout that.

Then forget about everything I said, I prefer grouped jobs and I understand now why it would be a hassle to do cleanup in `_execute_split_jobs`